unic/char/range/tests/iter_tests.rs

```diff
@@ -0,0 +1,63 @@
+extern crate unic_char_range;
+
+use std::{char, u32};
+use unic_char_range::*;
+
+fn all_chars() -> Vec<char> {
+    (u32::MIN..u32::MAX)
+        .take_while(|&u| u <= char::MAX as u32)
+        .filter_map(char::from_u32)
+        .collect()
+}
+
+#[test]
+fn iter_all_chars() {
+    assert!(CharRange::all().iter().eq(all_chars()));
+}
+
+#[test]
+fn iter_all_chars_rev() {
+    assert!(
+        CharRange::all()
+            .iter()
+            .rev()
+            .eq(all_chars().into_iter().rev())
+    )
+}
+
+#[test]
+fn iter_all_chars_mixed_next_back() {
+    let mut custom = CharRange::all().iter();
+    let mut simple = all_chars().into_iter();
+    while let Some(custom_char) = custom.next() {
+        assert_eq!(Some(custom_char), simple.next());
+        assert_eq!(custom.next_back(), simple.next_back());
+    }
+    assert_eq!(None, simple.next());
+}
+
+#[test]
+fn iter_fused() {
+    let mut iter = CharRange::all().iter();
+    let mut fused = all_chars().into_iter().fuse();
```

Comment:
[The fn to fuse an iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fuse) is stable. It's merely the trait to declare yourself as fused that is unstable.